### ISSUE TITLE
[suggestion] A full cauldron of dyed water can color an entire stack of items when thrown in together

### ISSUE THREAD

**Issue by stohun01** [asiekierka/CharsetMC#201]
Items clump together in stacks when in the world. I'm suggesting that when you throw a stack of items (all at once and all the same type) into a cauldron that is 100% filled with dyed water that the entire stack is dyed emptying the cauldron completely. It would give incentive to using a cauldron for dying over the crafting table.

**Comment by asiekierka** [comment]
Feels a bit hardcoded/edge-casey. Not something I'm interested in.

Currently, the idea is for cauldron-based dyeing to *replace* grid-based dyeing, by adding an opt-in option to disable the latter. Thus, they should be about on par - and that's what is the case now.

Unfortunately, fluids are not cleanly divisible by 64, so the "125mB per item" is probably here to stay, for now at least. I still want to look into the issues of storing and clearing/adding to dyed water, as the solutions there are now are not ideal.

**Comment by stohun01** [comment]
My point basically is that it takes more effort to dye a stack of something like wool blocks using a cauldron compared to the crafting grid. With a crafting grid it'd be around 4 clicks to make a stack of dyed wool if you used JEI. Using a cauldron you'd have to fill it with water then the dye then drop the blocks and then repeat that 7 more times for a whole stack. I'm suggesting that if you drop a single item in it behaves as it does currently, but if you drop a whole stack in (like drag and drop from your inventory) and it goes in as a single clump and the cauldron contains no less than 1000mB of dyed water that the result is the whole stack being dyed and using up all the dyed water. There would be no need to divide the fluid by 64.

**Comment by asiekierka** [comment]
Yes, but then if you insert 63 items, they would need 63*125mB still. I don't like that at exactly 64 items the amount of liquid required magically becomes 1000.

Perhaps for a stack of X items I could require 125*ceil(sqrt(X)) mB. That is, 125mB for 1 item, 250mB for 2-4 items, 375mB for 5-9 items, etc. That could work.
